### DISCUSSION TITLE
Strange behavior when using PixiJS 8 within a Vue.js component

### DISCUSSION THREAD

**Discussion by 2nOrderEDO** [pixijs/pixijs#10444] [Q&A] [answered]
Hello fellow forum users,

I'm facing some problems when trying to load pixi within a vue component. Just wanted to check if it could be a bug or just that I'm a complete n00b.

In order for pixi to be available within the component, I declare new PIXI.Application() within the data() segment of Vue. This works fine and I can even load a texture and display it within the appropriate HTML element I reserved for the canvas. In other words, I can replicate pixi.js tutorial 1 within a vue component without problems.

The problem I'm facing is when I try to add a plain, no fancy Text element. Then, I get an error saying that: "CanvasRenderingContext2D.createPattern: Argument 1 could not be converted to any of: HTMLImageElement, SVGImageElement, HTMLCanvasElement, HTMLVideoElement, OffscreenCanvas, ImageBitmap, VideoFrame."

This is in Firefox, in Brave I get: "Failed to execute 'createPattern' on 'CanvasRenderingContext2D': The provided value is not of type '(CSSImageValue or HTMLCanvasElement or HTMLImageElement or HTMLVideoElement or ImageBitmap or OffscreenCanvas or SVGImageElement or VideoFrame)'."

I will leave here my code for the Vue component for your reference:

```HTML
<template>
  <div ref="stage"></div>
</template>

<script>
  import * as PIXI from 'pixi.js';

  export default {
    data() {
      return {
        app: new PIXI.Application(), 
      };
    },
    mounted() {
      
      (async () => {
        
        await this.app.init({
          background: '#FFFFFF',
          resizeTo: HTMLElement,
          width: 1280,
          height: 720,
          antialias: true, 
          autoDensity: true 
        }); 

        this.$refs.stage.appendChild(this.app.canvas);

        const texture = await PIXI.Assets.load('https://pixijs.com/assets/bunny.png');
        const bunny = new PIXI.Sprite(texture);
        this.app.stage.addChild(bunny);

        bunny.x = this.app.screen.width / 2;
        bunny.y = this.app.screen.height / 2;

        const basicText = new PIXI.Text({ text: 'Hello World'});

        basicText.x = this.app.screen.width / 2;
        basicText.y = this.app.screen.height / 2; 

        this.app.stage.addChild(basicText);

      })();
    },
};
</script>

<style>
</style>
```

If you remove the line where basicText is added to the stage, you get no error. I'm using the latest version of pixi (8.1).

Thanks for your time reading my post and for providing any tips 🙂

BR,

2nOrderEDO

**Comment by sangyookm** [comment]
I get this too..

**Reply by jordanlevy96** [reply]
Same here, tried on Firefox and Chrome

**Comment by jordanlevy96** [comment]
FWIW, Bitmap Text does seem to work with Vue. Check out an example here: https://pixijs.com/8.x/examples/text/bitmap-text

And you can use this website to get fnt files: https://snowb.org/

**Reply by jordanlevy96** [reply]
I just got this same error (see below) when trying to add a dropShadow to the text settings (it still works without the dropShadow). Hopefully this info helps someone.

```typescript
...
const style = {
  fontFamily: 'Baskerville',
  fontSize: props.size || 24,
  tint: 0xffffff,
  dropShadow: true,
  dropShadowColor: '#000000',
  dropShadowBlur: 4,
  dropShadowAngle: Math.PI / 6,
  dropShadowDistance: 6,
}

onMounted(() => {
  const app = props.app

  bitmapFontText = new PIXI.BitmapText({
    text: props.text,
    style,
  })

  app.stage.addChild(bitmapFontText)
})
...
```

It is only the dropShadow value that seems to matter; if I comment it out, the text renders like before (i.e. with no dropShadow, so I guess it is ignoring the other values).

```
TypeError: Failed to execute 'createPattern' on 'CanvasRenderingContext2D': The provided value is not of type '(CSSImageValue or HTMLCanvasElement or HTMLImageElement or HTMLVideoElement or ImageBitmap or OffscreenCanvas or SVGImageElement or VideoFrame)'.
    at getCanvasFillStyle (getCanvasFillStyle.ts:22:33)
    at _DynamicBitmapFont2._setupContext (DynamicBitmapFont.ts:367:33)
    at _DynamicBitmapFont2._nextPage (DynamicBitmapFont.ts:319:14)
    at _DynamicBitmapFont2.ensureCharacters (DynamicBitmapFont.ts:132:29)
    at BitmapFontManagerClass.getFont (BitmapFontManager.ts:172:10)
    at Proxy._updateContext (BitmapTextPipe.ts:116:46)
    at Proxy.initGpuText (BitmapTextPipe.ts:193:14)
    at Proxy._getGpuBitmapText (BitmapTextPipe.ts:183:60)
    at Proxy.addRenderable (BitmapTextPipe.ts:57:41)
    at collectAllRenderablesSimple (buildInstructions.ts:75:36)
```

**Comment by yurijmikhalevich** [comment]
I see the same with Pixi.js 8.2.5, Vue.js, and Text.

**Reply by yurijmikhalevich** [reply]
FYI, after looking into the pixe.js code, I was able to resolve this by passing a `FillGradient` to `style.fill`. This looks like a bug with `pixi.js` to me.

**Reply by alcalyn** [reply]
Thanks, replacing

``` ts
const coordsTextStyle = new TextStyle({
    fill: textColor,
});
```

by this:

``` ts
const fill = new FillGradient(0, 0, 1, 1);
fill.addColorStop(0, textColor);
fill.addColorStop(1, textColor);

const coordsTextStyle = new TextStyle({
    fill,
});
```

fixed the error by creating text like this and trying to render it:

``` ts
new Text({ text: 'a', style: coordsTextStyle });
```

**Comment by nenjack** [comment]
have the same thing in latest pixi v8+ within react app

**Comment by rossinek** [comment]
Solution for Vue 3: store `app` in a `shallowRef` instead of `ref` ([docs](https://vuejs.org/api/reactivity-advanced.html#shallowref)). This fixed the problem for me.

**Reply by alcalyn** [reply, marked as answer]
Ah yes, better ! I'd be curious if there is a case where storing a vue app in a ref() is required.

Recently, after upgrade to pixi 8, I add to change all refs of vue app to raw instances to prevent lot of lags. Using shallowRef may have be better
